libvast/vast/bitmap_algorithms.hpp

```diff
@@ -419,26 +426,26 @@ class bitwise_range : public detail::range_facade<bitwise_range<BitRange>> {
     }
   }
 
-  /// Advances the range to a given position and then performs a subsequent
-  /// call to select().
-  /// @param x The ID where to start the call to select().
+  /// Selects the next bit of a given value starting at given position.
+  /// @param i The position to start  where to start the call to select().
+  /// @pre `!done() && x >= offset()`
   template <bool Bit = true>
-  void select_at(vast::id x) {
-    VAST_ASSERT(i_ != npos);
-    if (x < id())
-      return;
-    if (x > id()) {
-      next(x - id());
+  void select_at(id x) {
```

Comment:
Per our last discussion: this object behaves like a forward pointer. However, `..._at()` says index-based random access to me. Did you consider alternatives? Maybe `skip_until` would give a better intuition about the algorithm, i.e., it skips forward until given offset/threshold is reached.